cpp/terrain/terrain_object.h

```diff
@@ -108,12 +108,15 @@ class TerrainObject : public std::enable_shared_from_this<TerrainObject> {
 	bool is_floating() const;
 
 	/**
-	 * has the object been placed
+	 * returns true if this object has been placed. this indicates that the object has a position and exists
+	 * on the map, floating buildings are not considered placed as they are only an indicator
+	 * for where something can begin construction
```

Comment:
k.
maybe you'll want to document the list of all possible states.
while you're at it, just create an enum for all possible states, and add a function that returns the current state... maybe `is_placed` and `is_floating` would both get obsoleted by that?